jena-fuseki2/jena-fuseki-ui/src/views/dataset/Edit.vue

```diff
@@ -277,7 +278,8 @@ export default {
       if (!this.saveGraphDisabled) {
         this.loadingGraph = true
         try {
-          await this.$fusekiService.saveGraph(this.datasetName, this.selectedGraph, this.content)
+          const dataEndpoint = this.services['gsp-rw']['srv.endpoints'].find(endpoint => endpoint !== '') || ''
+          await this.$fusekiService.saveGraph(`${this.datasetName}/${dataEndpoint}`, this.selectedGraph, this.content)
           displayNotification(this, `Graph updated for dataset "${this.datasetName}"`)
```

Comment:
Note, the rest of the changes were added later to fix/add Cypress tests. This file is the only change to fix the issue mentioned in this pull request.